total_scattering/autogrouping/similarity.py

```diff
@@ -0,0 +1,90 @@
+# code it
+# Bundle all of this into a class, rename it something better
+import numpy
+
+'''
+This file calculates similarity metrics for numpy vectors.
+It is build from cross correlations
+'''
+
+
+class similarity_metric:
+
+    def __init__(self, L=5):
+
+        self.L = L
+        self.weights = [self.weight_triangular(r) for r in range(0, self.L + 1)]
+
+    '''
+    symmetric weight function
+    '''
+
+    def weight_triangular(self, r):
+
+        return (self.L - abs(r) + 1) / (self.L + 1)
+
+    '''
+    f, g ---> vectors to cross correlate.
+
+    L ---> From their initial aligned positions, vectors will be cross
+           correlated at offsets in union([ 0, L ] and [ 0, -L ] ), inclusive.
+
+    weights ---> weight[ i ] to be applied to cross_correlation[ i ]
+    for i in [ -L, L ] inclusive
+
+    returns ---> weighted cross correlation of the vectors:
+                 sum_over_r( weight[ r ] * sum_over_x ( f( x ) * g( x + r ) ) )
+    '''
```

Comment:
Do you mind to move the above comment so that it is a docstring under the `triangular_weighted_cross_correlation` function?